Framework/HistogramData/test/AddableTest.h

```diff
@@ -0,0 +1,92 @@
+#ifndef MANTID_HISTOGRAMDATA_ADDABLETEST_H_
+#define MANTID_HISTOGRAMDATA_ADDABLETEST_H_
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidHistogramData/Addable.h"
+#include "MantidHistogramData/FixedLengthVector.h"
+#include "MantidHistogramData/HistogramX.h"
+#include "MantidHistogramData/Iterable.h"
+#include "MantidHistogramData/VectorOf.h"
+
+using Mantid::HistogramData::detail::Addable;
+using Mantid::HistogramData::detail::FixedLengthVector;
+using Mantid::HistogramData::detail::Iterable;
+using Mantid::HistogramData::detail::VectorOf;
+using Mantid::HistogramData::HistogramX;
+
+class AddableTester : public VectorOf<AddableTester, HistogramX>,
```

Comment:
Well, in principle I could just test it with `std::vector<double>`, I think. We would then have no test of the functionality with `FixedLengthVector` and `VectorOf`, but basically that simply uses iterator access.